README.md

```diff
@@ -12,3 +12,31 @@ The two branches should be maintained as closely as possible.
 To deploy a custom buildfarm using this configuration you might want to use the
 [ros_buildfarm](https://github.com/ros-infrastructure/ros_buildfarm)
 repository.
+
+#Contributing guidelines
+
+## Branch hygiene
+Be careful about which branch you fork from and submit a pull request to.
+
+If you are contributing a configuration change (e.g. changing the values of keys) to be deployed on build.ros.org, make a pull request to the `production` branch.
+
+If you are contributing a change to be deployed on the testing farm or a more structural change, make a pull request to the `master` branch.
+
+##Guide to blacklisting packages
+
+If you wish to blacklist a package on build.ros.org, submit a pull request to `production` doing the following:
+
+Find the `release-build.yaml` file which is appropriate for the platform you wish to blacklist your package for under the folder for the appropriate distribution.
+
+For example, if you want to blacklist a package on 32-bit ARM in Kinetic Kame, you want to edit the file `kinetic/release-armhf.build`.
+
+Add the name of the package to blacklist under the entry `package_blacklist`, e.g.
+
+```
+  package_blacklist:
+    - my_blacklisted_package
+```
+
+When you submit a pull request, please *open a ticket* in `ros_buildfarm_config` or the repository for your package describing why you are blacklisting your package, and link to that ticket in your pull request to blacklist the package.
```

Comment:
The requested information could just be provided in the pull request description. Why does a user need to create two things, a ticket *and* a pull request?